sktime/forecasting/arch/_uarch.py

```diff
@@ -0,0 +1,362 @@
+        First observation to use when estimating model
+    last_obs : {int, str, datetime, Timestamp}
+        Last observation to use when estimating model
+    tol : float, optional
+        Tolerance for termination.
+    options : dict, optional
+        Options to pass to scipy.optimize.minimize.  Valid entries
+        include 'ftol', 'eps', 'disp', and 'maxiter'.
+    backcast : {float, np.ndarray}, optional
+        Value to use as backcast. Should be measure \\sigma^2_0
+        since model-specific non-linear transformations are applied to
+        value before computing the variance recursions.
+    params : {np.ndarray, Series}
+        Parameters required to forecast. Must be identical in shape to the
+        parameters computed by fitting the model.
+    start : {int, datetime, Timestamp, str}, optional
+        An integer, datetime or str indicating the first observation to produce the
+        forecast for. Datetimes can only be used with pandas inputs that have a
+        datetime index. Strings must be convertible to a date time, such as in
+        '1945-01-01'.
+    align : str, optional
+        Either 'origin' or 'target'. When set of 'origin', the t-th row of forecast
+        contains the forecasts for t+1, t+2, ..., t+h. When set to 'target', the
+        t-th row contains the 1-step ahead forecast from time t-1, the 2 step from
+        time t-2, ..., and the h-step from time t-h. 'target' simplified computing
+        forecast errors since the realization and h-step forecast are aligned.
+    method : {'analytic', 'simulation', 'bootstrap'}
+        Method to use when producing the forecast. The default is analytic.
+        The method only affects the variance forecast generation. Not all volatility
+        models support all methods.
+        In particular, volatility models that do not evolve in squares such as
+        EGARCH or TARCH
+        do not support the 'analytic' method for horizons > 1.
+    simulations : int
+        Number of simulations to run when computing the forecast using either
+        simulation or bootstrap.
+    rng : callable, optional
+        Custom random number generator to use in simulation-based forecasts.
+        Must produce random samples using the syntax rng(size) where size the
+        2-element tuple (simulations, horizon).
+    random_state : RandomState, optional
+        NumPy RandomState instance to use when method is 'bootstrap'
+    reindex : bool, optional
+        Whether to reindex the forecasts to have the same dimension as the series
+        being forecast. Prior to 4.18 this was the default. As of 4.19 this is now
+        optional. If not provided, a warning is raised about the future change in
+        the default which will occur after September 2021.
+
+    See Also
+    --------
+    StatsForecastARCH
+    StatsForecastGARCH
+
+    References
+    ----------
+    .. [1] Jason Brownlee. How to Model Volatility with ARCH and GARCH for Time Series
```

Comment:
This is a kind of surprising reference in a package documentation.

Reply:
I should have edited it, will change it to official doc reference, btw I worked on this after a very long time.